openexr: Add custom libs property

payerle opened the pull request:
Libraries for openexr are named libOpenEXR*.so, etc., so the default libs handler in spec does not find them.

Add a custom libs property to address this.

Partial fix for #42273

payerle committed:
openexr: Add custom libs property

Libraries for openexr are named libOpenEXR*.so, etc., so the default libs
handler in spec does not find them.

Add a custom libs property to address this.

Partial fix for #42273

alalazo commented on var/spack/repos/builtin/packages/openexr/package.py:
```diff
@@ -85,6 +85,11 @@ class Openexr(CMakePackage, AutotoolsPackage):
+    def libs(self):
+        # Override because libs have different case than Spack package name
+        name = "libOpenEXR*"
+        return find_libraries(name, root=self.prefix, shared=True, recursive=True)
```
If we know where libraries are being installed, it would be good to set recursive to `False`

payerle committed:
:openexr reworked custom libs method

Reworked custum libs method to remove recursive flag from find_libraries
as per suggestion by @alalazo re similar mod to qb3 (PR #42275)

alalazo commented:
@spackbot fix style

payerle committed:
[@spackbot] updating style on behalf of payerle
